### ISSUE TITLE
Configurations and Synoptics : Manager mode protection

### ISSUE THREAD

**Issue by GDH-ISIS** [ISISComputingGroup/IBEX#3489]
To protect the configurations and synoptics, should it be discussed whether the options to edit, delete and create these items should be safeguarded behind the manager mode option.

**Comment by kjwoodsISIS** [comment]
Yes, I think it is definitely worth asking this question.  I guess only the scientists can answer this question for us.  We can ask the SAG, if you like?

**Comment by ChrisM-S** [comment]
Are these "deleted" items still available in the git record of the configuration?
  - If not, should they be?
  - If they are, how much skill would be needed to recover them after accidental deletion e.g.
        - experienced software developer?
        - supporter of the week?
        - scientist with manager mode level of IBEX experience?

**Comment by Tom-Willemsen** [comment]
Yes, the deleted items will be in the configuration history. Recovering a deleted synoptic requires knowledge of `git log` (to find a commit from before it was deleted) and `git checkout` - so any developer should be able to do it.

**Comment by ChrisM-S** [comment]
If commits for configurations were tagged/branched? at the time of saving in a user understandable way (with something easy like a time string), could it be simple to provide a "revert to ...x... version" of a config/synoptic?  Easy recovery might be useful for a known mistake but could also reduce the impact of unwise changes/deletions by an inexperienced user allowing reversion to a previously working setup.  Even a diff might be helpful in diagnosing typos/mistakes in something like a port number macro.

**Comment by ThomasLohnert** [comment]
+1 from Muons

**Comment by John-Holt-Tessella** [comment]
PB confirms that this looks good (30/8/2019):

> Yes, that looks right to me. This should be sufficient to prevent all but intentional vandalism and captures the things we’d thought of in making the original suggestion.

**Comment by Tom-Willemsen** [comment]
Re-requested by NIMROD

**Comment by DominicOram** [comment]
@Tom-Willemsen and I just discussed that given how much of the system this touches it would be a good one for new starters (@RaiBishal maybe?) to get involved in. A senior dev should:
* Split this into manageable tickets for new starters (e.g. part 1 above could actually be a separate ticket)
* Do the part for getting the blockserver to adhere to manager mode
* Oversee work done by new starters

**Comment by Tom-Willemsen** [comment]
Split into 3 more manageable tickets:
- https://github.com/ISISComputingGroup/IBEX/issues/4794
- https://github.com/ISISComputingGroup/IBEX/issues/4795
- https://github.com/ISISComputingGroup/IBEX/issues/4796

**Comment by John-Holt-Tessella** [comment]
For review check that the functionality in this ticket is represented in the other 3 tickets.

**Comment by JamesKingWork** [comment]
Review: See comment on https://github.com/ISISComputingGroup/IBEX/issues/4795

> Review from https://github.com/ISISComputingGroup/IBEX/issues/3489: this ticket should specify that the user should not be able to save protected configs and components when not in manager mode, both through the GUI and through the PV
